apps/api/src/hyp3_api/auth.py

```diff
@@ -4,13 +4,29 @@
 import jwt
 
 
-def decode_token(token: str | None) -> dict | None:
-    if token is None:
-        return None
+class InvalidTokenException(Exception):
+    """Raised when authorization token cannot be decoded."""
+
+
+def decode_edl_bearer_token(token: str, jwks_client: jwt.PyJWKClient) -> tuple[str, str]:
+    signing_key = jwks_client.get_signing_key('edljwtpubkey_ops')
     try:
-        return jwt.decode(token, environ['AUTH_PUBLIC_KEY'], algorithms=environ['AUTH_ALGORITHM'])
-    except (jwt.ExpiredSignatureError, jwt.DecodeError):
-        return None
+        payload = jwt.decode(token, signing_key, algorithms=['RS256'])
+    except jwt.exceptions.InvalidTokenError as e:
+        raise InvalidTokenException(f'Invalid authorization token provided: {str(e)}')
+    return payload['uid'], token
+
+
+def decode_asf_cookie(cookie: str) -> tuple[str, str]:
+    try:
+        payload = jwt.decode(cookie, environ['AUTH_PUBLIC_KEY'], algorithms=environ['AUTH_ALGORITHM'])
+    except jwt.exceptions.InvalidTokenError as e:
+        raise InvalidTokenException(f'Invalid authorization cookie provided: {str(e)}')
```

Comment:
Are there any security concerns with exposing the error message directly from the `jwt` library by putting `str(e)` in the error message? I don't think there would be, since we're just doing public key decoding, but just want to make sure.